src/app.ts

```diff
@@ -110,6 +175,25 @@ export function buildApp(
 
 	app.use(jsonBodyParser());
 
+	const uiHandler = (req: express.Request, res: express.Response) => {
+		withPandaAuth(
+			getPanda,
+			req,
+			res,
+			() => {
+				res.contentType('html').send(getUI());
+			},
+			() => {
+				res.status(403)
+					.contentType('html')
+					.send(getRedirectResponse(req));
+			},
+		);
+	};
+
+	app.get('/', uiHandler);
+	app.get('/ui', uiHandler);
```

Comment:
I reckon we should put the UI at a single path (my preference would be at the root) - what do you think?